src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCachedSystemStoreProvider.cs

```diff
@@ -216,6 +217,12 @@ bool ProcessFile(string file, out DateTime lastModified, bool skipStat = false)
                     return false;
                 }
 
+                string originalFile = Interop.Sys.ReadLink(file) ?? file;
```

Comment:
This doesn't end up resolving the final path for links.
(And, if we use strings, we should specify the `StringComparison` on the `HashSet`)

Rather than storing the path strings, we can extend the `TryStatFile` method that gets called above so it returns as an `out` argument in a tuple the `long Ino` and `long Dev` from `struct FileStatus`.
This tuple can then be stored in the `processedFiles` `HashSet`.